src/components/Global/Header.js

```diff
@@ -0,0 +1,91 @@
+// dependencias
+import React, { Component } from 'react';
+import firebase from 'firebase';
+// import PropTypes from 'prop-types';
+// assets
+import logo from './images/logo.png';
+import cart from './images/carrito.png';
+import './css/Header.css';
+
+class Header extends Component {
+  constructor() {
+    super();// heredando de otra clase,  super llama al  header hereda de component, // state definir el estado de nuestro componente
+    this.state = {
+      user: null
+    };
+    // Al utilizar el objeto this realizar un bindeo
+    this.handleAuth = this.handleAuth.bind(this);
+    this.handleLogout = this.handleLogout.bind(this);
+    this.renderLoginButton = this.renderLoginButton.bind(this);
+  }
+
+  componentWillMount() { // este metodo es un ciclo de vida, se dispara una vez en componente ha sifo renderizado
+    firebase.auth().onAuthStateChanged(user => {
+      // devuelve un objeto usuario  , cada evz que salgamos o entramos tendra la info el usuario
+      this.setState({ user });
+    });
+  }
+
+  handleAuth() {
+    const provider = new firebase.auth.GoogleAuthProvider();
+    // proveedor de google
+
+    firebase.auth().signInWithPopup(provider)
+    // este sgning devuelve una promesa
+      .then(result => console.log(`${result.user.email} ha iniciado sesión `))
+      .catch(error => console.log(`Error ${error.code}: ${error.message}`));
+  }
+
+  handleLogout() {
+    firebase.auth().signOut()
+      .then(result => console.log(`${result.user.email} ha salido `))
+      .catch(error => console.log(`Error ${error.code}: ${error.message}`));
+  }
+
+  renderLoginButton() {
+    if (this.state.user) {
```

Comment:
Tal vez una buena opción sea extraer el usuario desde `this.state` ya que se ocupa varias veces en este método:

`const { user } = this.state`;